wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/rapidreactcommandbot/RapidReactCommandBot.java

```diff
@@ -0,0 +1,95 @@
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj.examples.rapidreactcommandbot;
+
+import static edu.wpi.first.wpilibj2.command.CommandGroupBase.parallel;
+
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.Constants.AutoConstants;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.Constants.OIConstants;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.Constants.ShooterConstants;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.subsystems.Drive;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.subsystems.Intake;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.subsystems.Shooter;
+import edu.wpi.first.wpilibj.examples.rapidreactcommandbot.subsystems.Storage;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.button.CommandJoystick;
+import edu.wpi.first.wpilibj2.command.button.CommandXboxController;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+
+/**
+ * This class is where the bulk of the robot should be declared. Since Command-based is a
+ * "declarative" paradigm, very little robot logic should actually be handled in the {@link Robot}
+ * periodic methods (other than the scheduler calls). Instead, the structure of the robot (including
+ * subsystems, commands, and button mappings) should be declared here.
+ */
+public class RapidReactCommandBot {
+  // The robot's subsystems
+  private final Drive m_drive = new Drive();
+  private final Intake m_intake = new Intake();
+  private final Storage m_storage = new Storage();
+  private final Shooter m_shooter = new Shooter();
+
+  // The driver's controller
+  CommandXboxController m_driverController =
+      new CommandXboxController(OIConstants.kDriverControllerPort);
+
+  /**
+   * Use this method to define non-button Event bindings. These are useful for automating robot
+   * behaviors based on sensor input.
+   *
+   * <p>Called during {@link Robot#robotInit()}.
+   *
+   * <p>Event binding methods are available on the {@link Trigger} class.
+   */
+  public void configureEventBindings() {
+    // Automatically run the storage motor whenever the ball storage is not full,
+    // and turn it off whenever it fills.
+    new Trigger(m_storage::isFull).negate().whileActiveContinuous(m_storage.runCommand());
+
+    // Automatically disable and retract the intake whenever the ball storage is full.
+    new Trigger(m_storage::isFull).whenActive(m_intake.retractCommand());
+  }
+
+  /**
+   * Use this method to define your Button bindings for teleoperated control.
+   *
+   * <p>Called during {@link Robot#teleopInit()}.
+   *
+   * <p>Button binding methods are available on the {@link CommandXboxController} and {@link
+   * CommandJoystick}
+   */
+  public void configureControllerBindings() {
+    // Control the drive with split-stick arcade controls
+    m_drive.setDefaultCommand(
```

Comment:
This being here could be confusing, because there are some default commands that do need to be bound at startup. Differentiating between the two isn't trivial, especially for inexperienced teams.
I'm not sure what the solution is.

Reply:
I don't think there's a one-size-fits-all solution.  It does depend on the mechanism and on coder preference.